lib/provider/aws/lambda-edge-origin-request/create-request.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const crypto = require('crypto');
+const Request = require('../../../request');
+
+function requestHeaders(event) {
+  let headers = Object.keys(event.request.headers).reduce((headers, key) => {
+    headers[event.request.headers[key][0].key.toLowerCase()] = event.request.headers[key][0].value;
+    return headers;
+  }, {});
+
+  headers['x-request-id'] = crypto.randomBytes(30).toString('base64')
+
+  return headers;
+}
+
+function requestBody(event) {
+  const type = typeof event.request.body;
+  const body = event && event.request && event.request.body;
+
+  if (!body) return '';
+
+  if (Buffer.isBuffer(event.request.body.data)) {
+    return event.request.body.data;
+  } else if (type === 'string') {
+    return Buffer.from(event.request.body.data, event.request.body.encoding === 'base64' ? 'base64' : 'utf8');
+  } else if (type === 'object') {
+    return Buffer.from(JSON.stringify(event.request.body.data));
```

Comment:
This is adding extra `""` around the returned request body.

Instead, this should be:
```
Buffer.from(event.request.body.data, "base64");
```

Reply:
Actually, it should be as follows (I've just seen how the original code is for APIG, so I can see how this got copy-and-pasted like this).

AFAIK it should be:

```javascript
function requestBody(event) {
  const body = event && event.request && event.request.body && event.request.body.data;
  const type = typeof body;

  if (!body) return '';

  if (Buffer.isBuffer(body)) {
    return body;
  } else if (type === 'string') {
    return Buffer.from(body, event.request.body.encoding === 'base64' ? 'base64' : 'utf8');
  } else if (type === 'object') {
    return Buffer.from(JSON.stringify(body));
  }

  throw new Error(`Unexpected event.body type: ${typeof event.request.body.data}`);
}
```

Reply:
I can confirm this works 👍

Reply:
Hey, thanks for the feedback, how is this different from the current implementation? In all cases, I will update the PR to use the body variable (more readable I guess)

Reply:
The current implementation results in the request body being wrapped in `"`, so if you have a request body of `"hello"`, with the previous code, you'd get a string of `"\"hello\""`.

AFAIK it's because you copied it from the APIG handler, which expects the body to be a string OR an already-decoded JSON object (and in the latter case, is why it uses a `JSON.stringify` -- to turn it back to a string).

In our case, for Edge, the body is always a string (either base64 encoded or not).